string-manipulation-2/src/truncate.js

```diff
@@ -0,0 +1,20 @@
+/* exported truncate */
+
+// We need a storage for the new string.
+// We need to check each character in string and first make sure a character exists in each position of the length.
+// We need to store the found characters in the new string.
+// We need to add ... to the new string.
+// We need to return the new string.
+
+function truncate(length, string) {
+  var newString = '';
+  for (var i = 0; i < length; i++) {
+    if (string[i] !== undefined) {
```

Comment:
If the `length` parameter is a huge number, this will end up doing a lot of extra operations. Consider if your string is 4 characters long, but your user requests a length of 1000. That is 9996 extra operations. Instead, you should check if the length requested is longer than the string, and if so, just work with the string itself.